README update (#380)
IntelPython/daal4py: README.md

- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py) when applied to scikit-learn from PyPi
- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master) when applied to build from master branch

> Core functioanlity of daal4py is in place Scikit-learn patching - Same Code, Same Behavior but faster execution. 

Reviewer: I prefer to add info how to download daal4py, because currently it's only how to build daal4py.
# Installation
daal4py can be installed from conda:
```bash
conda install daal4py -c intel
```

Author: done

Stock Scikit-learn

> ```

Reviewer: \```py to highlight python synaxis

Author: done

- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py) when applied to scikit-learn from PyPi
- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master) when applied to build from master branch

Core functioanlity of daal4py is in place Scikit-learn patching - Same Code, Same Behavior but faster execution. 

Stock Scikit-learn
```
from sklearn.svm import SVC

> X, Y = get_dataset()

Reviewer: Probably, it's better to replace this with:
```py
from sklearn.datasets import load_digits
digits = load_digits()
X, y = digits.data, digits.target
```
The same how we do it in oneDAL's README to allow to run the code by copy-past only.

Author: done

- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py) when applied to scikit-learn from PyPi
- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master) when applied to build from master branch

Core functioanlity of daal4py is in place Scikit-learn patching - Same Code, Same Behavior but faster execution. 

Stock Scikit-learn
```py
from sklearn.svm import SVC
from sklearn.datasets import load_digits
digits = load_digits()
X, y = digits.data, digits.target
clf = SVC().fit(X, y)
res = clf.predict(X)
```

Intel CPU optimizations patching
```py
import daal4py as d4p
d4p.patch_sklearn()
from sklearn.svm import SVC
from sklearn.datasets import load_digits
digits = load_digits()
X, y = digits.data, digits.target
clf = SVC().fit(X, y)
res = clf.predict(X)
```

Intel CPU/GPU optimizations patching
```py
import daal4py as d4p
from daal4py.oneapi import sycl_context
d4p.patch_sklearn()
from sklearn.svm import SVC
from sklearn.datasets import load_digits
digits = load_digits()
X, y = digits.data, digits.target

> with sycl_context(“gpu”):

Reviewer: Better to use correct correct " symbols:
```py
with sycl_context("gpu"):
```

kmeans_init(data, 10, t_method="plusPlusDense", distributed=True)
```
# Getting Started

> daal4py is easily built from source with the majority of the necessary prerequisites available on conda.  The instructions below detail how to gather the prerequisites, setting one's build environment, and finally building and installing the completed package.  daal4py can be built for all three major platforms (Windows, Linux, macOS). Multi-node (distributed) and streaming support can be disabled if desired. There is Intel GPU support avaialbe as well

Reviewer 2: ```suggestion
daal4py is easily built from sources with the majority of the necessary prerequisites available on conda. The instructions below detail how to gather the prerequisites, set your build environment, and finally build and install the completed package. daal4py can be built for all three major platforms (Windows, Linux, macOS). Multi-node (distributed) and streaming support can be disabled if needed. There is Intel GPU support available as well.
```

* Python version >= 3.6
* conda-build version >= 3
* C++ compiler with C++11 support

> For oneAPI support:

Reviewer 2: you need an extra line before this line

Author: fixed

* A DPC++ compiler 

> * Intel(R) oneAPI Data Analytics Library version 2021.1 or later (https://github.com/01org/daal)

Reviewer 2: update link to github?

Author: Updated

>   * You can use the pre-built conda package from Intel channel on anaconda.org (see below)

Reviewer 2: link to the section below?

Author: removed "see below" notice

## Building daal4py

> For build with oneAPI support additional step requered:

Reviewer 2: ```suggestion
To build with oneAPI support, additional step are required:
```

> 1. DPC++ compiler - DPCPPROOT variable should be defiend and pointed to DPCPP compier

Reviewer 2: ```suggestion
1. Point to DPC++ compiler by defining ``DPCPPROOT`` variable.
```

> 2. Intel(R) oneDAL with oneAPI support
>     a. Can be installed from Conda channel
>     b. Can be installed fron oneAPI packages repository and passed via DALROOT variable

Reviewer 2: also maybe add a link to conda?

not clear: if installed with conda, DALROOT is not needed?

```suggestion
2. Install Intel(R) oneAPI Data Analytics Library with oneAPI support:

    a. From Conda channel.
    b. From oneAPI packages repository (pass the path to oneDAL via DALROOT variable)
```

```
export DPCPPROOT=/opt/intel/oneapi/compiler/latest
export DALROOT=/opt/intel/oneapi/daal/latest

> ```

Reviewer 2: I'd split this code block and insert the relevant line after each item in the list above

Author: done

```
conda create -n DAAL4PY python=3.7 impi-devel daal daal-include cython jinja2 numpy clang-tools -c intel -c conda-forge
conda activate DAAL4PY
export DALROOT=$CONDA_PREFIX
export MPIROOT=$CONDA_PREFIX
```
For build with oneAPI support additional step requered:
1. DPC++ compiler - DPCPPROOT variable should be defiend and pointed to DPCPP compier
2. Intel(R) oneDAL with oneAPI support
    a. Can be installed from Conda channel

>     b. Can be installed fron oneAPI packages repository and passed via DALROOT variable

Reviewer 2: this repeats the same info as above, do you really need to repeat it?

Author: done

If building in High Sierra or higher, one may have to run into C++ build errors related to platform targets. Utilize ```export MACOSX_DEPLOYMENT_TARGET="10.9"``` if running into platform target issues.

## Building daal4py

> Requires Intel(R) oneAPI Data Analytics Library and MPI being properly setup, e.g. DALROOT and MPIROOT being set.

Reviewer 2: ```suggestion
Requires Intel(R) oneAPI Data Analytics Library and Intel(R) MPI Library being properly set up, meaning you have to set DALROOT and MPIROOT variables.
```

```
cd <checkout-dir>
python setup.py build_ext
```

## Installing daal4py

> Requires Intel(R) oneAPI Data Analytics Library and MPI being properly setup, e.g. DALROOT and MPIROOT being set.

Reviewer 2: same comment as above

Author: done
